OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -441,5 +452,131 @@ public void TestGetVariationWithBucketingId()
+        }
+
+        // Should return the variation the user is bucketed into when the user is bucketed into one of the experiments
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenMutexGroupAndUserIsBucketed()
+        {
+            var mutexExperiment = ProjectConfig.GetExperimentFromKey("group_experiment_1");
+            var expectedVariation = mutexExperiment.Variations[0];
+
+            DecisionServiceMock.Setup(ds => ds.GetVariation(ProjectConfig.GetExperimentFromKey("group_experiment_1"), "user1", new UserAttributes())).Returns(expectedVariation);
+
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_feature");
+            var variation = DecisionServiceMock.Object.GetVariationForFeatureExperiment(featureFlag, "user1", new UserAttributes());
+
+            Assert.AreEqual(expectedVariation, variation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is bucketed into experiment \"group_experiment_1\" of feature \"boolean_feature\"."));
+        }
+
+        // Should return null and log a message when the user is not bucketed into any of the mutex experiments 
+        [Test]
+        public void TestGetVariationForFeatureExperimentGivenMutexGroupAndUserNotBucketed()
+        {
+            var mutexExperiment = ProjectConfig.GetExperimentFromKey("group_experiment_1");
+            
+            DecisionServiceMock.Setup(ds => ds.GetVariation(It.IsAny<Experiment>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns<Variation>(null);
+
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_feature");
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeatureExperiment(featureFlag, "user1", new UserAttributes());
+
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO, "The user \"user1\" is not bucketed into any of the experiments on the feature \"boolean_feature\"."));
+        }
+
+        // Should return the bucketed experiment and variation 
+        [Test]
+        public void TestGetVariationForFeatureWhenTheUserIsBucketedIntoFeatureExperiment()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("string_single_variable_feature");
+            var expectedExperimentId = featureFlag.ExperimentIds[0];
+            var expectedExperiment = ProjectConfig.GetExperimentFromId(expectedExperimentId);
+            var expectedVariation = expectedExperiment.Variations[0];
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeatureExperiment(It.IsAny<FeatureFlag>(), It.IsAny<string>(), It.IsAny<UserAttributes>())).Returns(expectedVariation);
+
+            var actualVariation = DecisionServiceMock.Object.GetVariationForFeature(featureFlag, "user1", new UserAttributes());
+
+            Assert.AreEqual(expectedVariation, actualVariation);
```

Comment:
shouldn't we log message, assert log message.

Reply:
GetVariationForFeatureExperiment() is mocked that has been called from GetVariationForFeature()